src/main/java/org/pac4j/dropwizard/Pac4jConfiguration.java

```diff
@@ -1,7 +1,172 @@
 package org.pac4j.dropwizard;
 
-import io.dropwizard.Configuration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
-public interface Pac4jConfiguration<T extends Configuration> {
-    Pac4jFactory getPac4jFactory(T configuration);
+import org.pac4j.core.authorization.authorizer.Authorizer;
+import org.pac4j.core.authorization.generator.AuthorizationGenerator;
+import org.pac4j.core.client.BaseClient;
+import org.pac4j.core.client.Client;
+import org.pac4j.core.client.Clients;
+import org.pac4j.core.config.Config;
+import org.pac4j.core.matching.Matcher;
+import org.pac4j.jax.rs.filter.SecurityFilter;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * An interface defining getters for parameters required to configure a {@link SecurityFilter}.
+ */
+public class Pac4jConfiguration {
```

Comment:
I think I'm missing something. Why exactly do you want users to be able to modify the config instance within `run` methods? The `Pac4jBundle#run` is invoked before the application's `run` method, so the Pac4j configuration options have already been applied by the time you get to application code.

More generally, I don't think modifying configuration at runtime is something we want to encourage. It's of course _possible_ (via config factories' setter methods), but it seems to me that it goes against the premise of out-of-process configuration which Dropwizard espouses (i.e. config by file or environment variables). Applications ingest configuration from some external source (which is easily editable by system administrators) rather than opaquely making decisions for themselves.